src/core/eme/eme_events.ts

```diff
@@ -0,0 +1,168 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IMediaKeySession } from "../../compat";
+
+type LicenseObject =
+  BufferSource |
+  ArrayBuffer |
+  ArrayBufferView;
+
+type sessionManagementEvents = "reuse-loaded-session" | "loaded-persistent-session";
+type sessionCreationEvents = "created-temporary-session" | "created-persistent-session";
+type mediaKeyMessageEvents =
+  "license-request" |
+  "license-renewal" |
+  "license-release" |
+  "individualization-request" |
+  "key-status-change";
+
+interface ISessionRequestEvent {
+  type : "generated-request";
+  value : {
+    session: IMediaKeySession|MediaKeySession;
+    sessionInfos: {
+      initData: Uint8Array;
+      initDataType: string;
+    };
+  };
+}
+
+interface ISessionManagementEvent {
+  type : sessionManagementEvents;
+  value : {
+    session : IMediaKeySession|MediaKeySession;
+    sessionInfos: {
+      storedSessionId?: string;
+    };
+  };
+}
+
+interface IMediaKeyMessageEvent {
+  type : mediaKeyMessageEvents;
+  value : {
+    session : IMediaKeySession|MediaKeySession;
+    sessionInfos: {
+      license: LicenseObject;
+    };
+  };
+}
+
+interface ISessionCreationEvent {
+  type: sessionCreationEvents;
+  value: {
+    session: IMediaKeySession|MediaKeySession;
+    sessionInfos: {
+      initData: Uint8Array;
+      initDataType: string;
+    };
+  };
+}
+
+type ISessionEvent =
+  ISessionManagementEvent |
+  IMediaKeyMessageEvent |
+  ISessionCreationEvent |
+  ISessionRequestEvent;
+
+/**
+ * Create the Object emitted by the EME Observable.
+ * @param {string} type - type of the event
+ * @param {MediaKeySession} session - MediaKeySession concerned
+ * @param {string} storedSessionId - Supplementary data, will be merged with the
+ * session information in the returned object.
+ * @returns {Object}
+ */
+function sessionManagementEvent(
+  type: sessionManagementEvents,
+  session: IMediaKeySession|MediaKeySession,
+  storedSessionId?: string
+): ISessionManagementEvent {
+  return {
+    type,
+    value: {
+      session,
+      sessionInfos: {
+        storedSessionId,
+      },
+    },
+  };
+}
+
+function sessionCreationEvent(
+  type: sessionCreationEvents,
+  session: IMediaKeySession|MediaKeySession,
+  initData: Uint8Array,
+  initDataType: string
+): ISessionCreationEvent {
+  return {
+    type,
+    value: {
+      session,
+      sessionInfos: {
+        initData,
+        initDataType,
+      },
+    },
+  };
+}
+
+function sessionRequestEvent(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update this function so that its implementation is not identical to the one on line 105. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=187&issues=AWLjaM-Xjr7wspRo8NBN&open=AWLjaM-Xjr7wspRo8NBN)

[ISSUE_KEY]: AWLjaM-Xjr7wspRo8NBN